system/jlib/jthread.cpp

```diff
@@ -1899,6 +1899,8 @@ protected: friend class PipeWriterThread;
         if (haserror)
             if (::pipe(errpipe)==-1)
                 throw makeOsException(errno);
+        unsigned argc;
+        char **argv=splitargs(prog,argc);
```

Comment:
Worth putting a comment about why this is done before the fork, or some bright spark will think they can optimize it in the future.